src/idl_parser.cpp

```diff
@@ -590,12 +590,33 @@ CheckedError Parser::ParseType(Type &type) {
     NEXT();
     Type subtype;
     ECHECK(Recurse([&]() { return ParseType(subtype); }));
-    if (subtype.base_type == BASE_TYPE_VECTOR) {
+    if (subtype.base_type == BASE_TYPE_VECTOR ||
+        subtype.base_type == BASE_TYPE_ARRAY) {
       // We could support this, but it will complicate things, and it's
       // easier to work around with a struct around the inner vector.
-      return Error("nested vector types not supported (wrap in table first).");
+      return Error("nested vector types not supported (wrap in table first)");
+    }
+    if (token_ == ':') {
+      if (!IsScalar(subtype.base_type)) {
+        return Error(
+              "arrays of non-scalar types not supported");
+      }
+      NEXT();
+      if (token_ != kTokenIntegerConstant) {
+        return Error(
+              "length of fixed-length array must be an integer value");
+      }
+      int64_t fixed_length = StringToInt(attribute_.c_str());
+      if (fixed_length < 1 || fixed_length > 0x7fff) {
```

Comment:
https://stackoverflow.com/a/7724909 says 2^16-1 is a C limit on object size, so it would be reasonable to limit the size to 16 bits, though I'm not sure about a 2^15-1 length limit. I'd rather place a limit on struct sizes of 2^16-1 and forego array length limits.